fiona/ogrext.pyx

```diff
@@ -1238,6 +1238,15 @@ cdef class Iterator:
         else:
             OGR_L_SetSpatialFilter(cogr_layer, NULL)
 
+        if where:
+            where_b = where.encode('utf-8')
+            where_c = where_b
+            exc_wrap_int(
+                OGR_L_SetAttributeFilter(cogr_layer, <const char*>where_c))
```

Comment:
@mwtoews let's catch that CPLEx exception here and raise something like fiona.errors.AttributeFilterError. In general, I'd like this project improve in the direction of only intentionally raising its own errors, with Python or GDAL errors being a signal of buggy code.